### ISSUE TITLE
[Data Issue]: US_PREPA.py

### ISSUE THREAD

**Issue by irmajavi** [electricitymaps/electricitymaps-contrib#4792]
### When did this happen?

2022-11-21

### What zones are affected?

US-pr Puerto Rico

### What is the problem?

The data source for parser US_PREPA.py is not returning any value. I think how the data is collected must be change because in the aeepr.com page the data is display correctly

**Comment by VIKTORVAV99** [comment]
I'll take a look later today. Thanks for reporting the issue.

**Comment by madsnedergaard** [comment]
Here's the error logs to help debugging :)

https://storage.googleapis.com/electricitymap-parser-logs/PR.html

<img width="762" alt="Screenshot 2022-11-21 at 21 22 57" src="https://user-images.githubusercontent.com/3296643/203151231-27b2d1fd-2f46-45a8-8709-1db426526e0b.png">

**Comment by VIKTORVAV99** [comment]
> Here's the error logs to help debugging :) Also note that this zone is geo-blocked (only accessible from Americas), so testing it is a bit tricky - if someone located in the US could help with this that would be great (unless you've already looked into it @VIKTORVAV99)
> 
> https://storage.googleapis.com/electricitymap-parser-logs/PR.html
> 
> <img alt="Screenshot 2022-11-21 at 21 22 57" width="762" src="https://user-images.githubusercontent.com/3296643/203151231-27b2d1fd-2f46-45a8-8709-1db426526e0b.png">

I did look into it a little and I got as far as encoding issues but I don't know what the real issue is just yet as it seems to work just fine when accessing the endpoint directly via the proxy and decoding the gzip file.

Now I haven't determined if this is related just yet or if it's just a local issue I have so if anyone want to take a peak at it they are more than welcome.

**Comment by VIKTORVAV99** [comment]
Just a small update:
Found a way to get the data but the parser still didn't work so I'm rewriting it in a type-safe manner that should be easier to debug in the future.

**Comment by VIKTORVAV99** [comment]
Update 2:
Parser is working for fetching the production but I can't seem to find a reliable datetime, the old parser also had this issue and it was not clear if that datetime updated the same time as the production figures.

@irmajavi could you check if there is a datetime anywhere on the site that seem to indicate when the data was updated?

**Comment by irmajavi** [comment]
I verify the site and in this page it shows the date.
I think the page to get the data needs to be change. In this page it shows
different shotcuts to various resources.

https://aeepr.com/es-pr/generaci%C3%B3n

**Comment by VIKTORVAV99** [comment]
After digging into the HTML a bit I can see that it says when that page was last updated but there seems to be no way to tie this to when the data was updated and I can't see that they have set a specific publishing time or delays anywhere.

So as it stands now I don't think we can use this particular source because we have no way of verifying the data timestamp itself. This means might show things on a vastly different time period than it actually represents and I don't think we want to take that risk.

**Comment by q--** [comment]
I already commented on the issue 

https://github.com/electricitymaps/electricitymaps-contrib/issues/3999#issuecomment-1127973014

TL;DR it appears the data source is no longer getting updated, and the new website didn't have the data we need at the time I checked

**Comment by q--** [comment]
I checked again, and it looks like the new website has generation data now: https://lumapr.com/system-overview/?lang=en

Please note that they're using geoblocking, so if you're not in the United States you'll need to use a VPN or similar to view it.

**Comment by mathilde-daugy** [comment]
Closing this issue as there is issue [#3999 ](https://github.com/electricitymaps/electricitymaps-contrib/issues/3999) on the same topic
